### PR TITLE
Added package cyanrip to rips cds via otg connected cd/dvd player

### PR DESCRIPTION
**Pull request by tathastu871**
Added package cyanrip to rips cds via otg connected cd/dvd player and its dependencies libcdio, libcdio-paranoia. 
It also depends on  libmusicbrianz  whose PR is made on termux main repo.

### CODE REVIEW

File: packages/libcdio/build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
+}
+termux_step_make(){
+	cd ${TERMUX_PKG_SRCDIR} && make -j8
+}
+termux_step_make_install() {
+	cd ${TERMUX_PKG_SRCDIR} && make install
+}
+termux_step_post_make_install() {
+	echo 'To install python bindings use  pip3 install pycdio (Requires swig to be installed)'
```
Comment (Grimler91) [inline packages/libcdio/build.sh:21]: This should probably be echoed to user in a postinst script (see for example [texlive's postinst script](https://github.com/termux/termux-packages/blob/5b2a9e6295a5c2dde37289497ac508778fdf86f1/packages/texlive/build.sh#L62))

File: packages/libcdio/build.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_SUGGESTS="libcdio-paranoia, swig"
+termux_step_configure() {
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
+}
+termux_step_make(){
+	cd ${TERMUX_PKG_SRCDIR} && make -j8
+}
+termux_step_make_install() {
+	cd ${TERMUX_PKG_SRCDIR} && make install
```
Comment (Grimler91) [inline packages/libcdio/build.sh:18]: Should be no reason to override termux_step_make_install

File: packages/libcdio/build.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-${TERMUX_PKG_VERSION}.tar.bz2
+TERMUX_PKG_SHA256=8550e9589dbd594bfac93b81ecf129b1dc9d0d51e90f9696f1b2f9b2af32712b
+TERMUX_INSTALL_DEPS=true
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_SUGGESTS="libcdio-paranoia, swig"
+termux_step_configure() {
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
+}
+termux_step_make(){
+	cd ${TERMUX_PKG_SRCDIR} && make -j8
```
Comment (Grimler91) [inline packages/libcdio/build.sh:15]: Should be no reason to override termux_step_make

File: packages/libcdio/build.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
+TERMUX_PKG_VERSION=2.1.0
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-${TERMUX_PKG_VERSION}.tar.bz2
+TERMUX_PKG_SHA256=8550e9589dbd594bfac93b81ecf129b1dc9d0d51e90f9696f1b2f9b2af32712b
+TERMUX_INSTALL_DEPS=true
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_SUGGESTS="libcdio-paranoia, swig"
+termux_step_configure() {
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
```
Comment (Grimler91) [inline packages/libcdio/build.sh:12]: Should be no reason to override termux_step_configure, move the LDFLAGS fix to termux_step_pre_configure (see for example how [tasksh does it](https://github.com/termux/termux-packages/blob/6a86fd8b28121df27ab0e0861e3b7bd14273f680/packages/tasksh/build.sh#L12)).
libandroid-glob also needs to be a dependency

File: packages/libcdio/build.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/libcdio
+TERMUX_PKG_DESCRIPTION="The GNU Compact Disc Input and Control library (libcdio) contains a library for CD-ROM and CD image access. Applications wishing to be oblivious of the OS- and device-dependent properties of a CD-ROM or of the specific details of various CD-image formats may benefit from using this library."
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
+TERMUX_PKG_VERSION=2.1.0
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-${TERMUX_PKG_VERSION}.tar.bz2
+TERMUX_PKG_SHA256=8550e9589dbd594bfac93b81ecf129b1dc9d0d51e90f9696f1b2f9b2af32712b
+TERMUX_INSTALL_DEPS=true
+TERMUX_PKG_BUILD_IN_SRC=true
```
Comment (Grimler91) [inline packages/libcdio/build.sh:9]: Have you tried building out of source?

File: packages/libcdio/build.sh [hunk 6]
```diff
@@ -0,0 +1,23 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/libcdio
+TERMUX_PKG_DESCRIPTION="The GNU Compact Disc Input and Control library (libcdio) contains a library for CD-ROM and CD image access. Applications wishing to be oblivious of the OS- and device-dependent properties of a CD-ROM or of the specific details of various CD-image formats may benefit from using this library."
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
+TERMUX_PKG_VERSION=2.1.0
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-${TERMUX_PKG_VERSION}.tar.bz2
+TERMUX_PKG_SHA256=8550e9589dbd594bfac93b81ecf129b1dc9d0d51e90f9696f1b2f9b2af32712b
+TERMUX_INSTALL_DEPS=true
```
Comment (Grimler91) [inline packages/libcdio/build.sh:8]: This is not a variable packagers should set

File: packages/libcdio-paranoia/build.sh [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_INSTALL_DEPS=true
+TERMUX_PKG_BUILD_IN_SRC=true
+termux_step_configure() {
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
+}
+termux_step_make(){
+	cd ${TERMUX_PKG_SRCDIR} && make -j8
+}
+termux_step_make_install() {
+	cd ${TERMUX_PKG_SRCDIR} && make install
```
Comment (Grimler91) [inline packages/libcdio-paranoia/build.sh:18]: Should be no reason to override termux_step_make_install

File: packages/libcdio-paranoia/build.sh [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-paranoia-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=28d7d00e4a83d0221acda0fd2eb3e3240bf094db4c00a85998922201939fa952
+TERMUX_PKG_DEPENDS="libcdio"
+TERMUX_INSTALL_DEPS=true
+TERMUX_PKG_BUILD_IN_SRC=true
+termux_step_configure() {
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
+}
+termux_step_make(){
+	cd ${TERMUX_PKG_SRCDIR} && make -j8
```
Comment (Grimler91) [inline packages/libcdio-paranoia/build.sh:15]: Should be no reason to override termux_step_make

File: packages/libcdio-paranoia/build.sh [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
+TERMUX_PKG_VERSION=10.2+2.0.1
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-paranoia-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=28d7d00e4a83d0221acda0fd2eb3e3240bf094db4c00a85998922201939fa952
+TERMUX_PKG_DEPENDS="libcdio"
+TERMUX_INSTALL_DEPS=true
+TERMUX_PKG_BUILD_IN_SRC=true
+termux_step_configure() {
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
```
Comment (Grimler91) [inline packages/libcdio-paranoia/build.sh:12]: Should be no reason to override termux_step_configure, move export of -landroid-glob to termux_step_pre_configure instead

File: packages/libcdio-paranoia/build.sh [hunk 10]
```diff
@@ -0,0 +1,20 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/libcdio
+TERMUX_PKG_DESCRIPTION="This is a port of xiph.org's cdda paranoia to use libcdio for CDROM access."
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
+TERMUX_PKG_VERSION=10.2+2.0.1
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-paranoia-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=28d7d00e4a83d0221acda0fd2eb3e3240bf094db4c00a85998922201939fa952
+TERMUX_PKG_DEPENDS="libcdio"
+TERMUX_INSTALL_DEPS=true
+TERMUX_PKG_BUILD_IN_SRC=true
```
Comment (Grimler91) [inline packages/libcdio-paranoia/build.sh:10]: Have you tried building out of src?

File: packages/libcdio-paranoia/build.sh [hunk 11]
```diff
@@ -0,0 +1,20 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/libcdio
+TERMUX_PKG_DESCRIPTION="This is a port of xiph.org's cdda paranoia to use libcdio for CDROM access."
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
+TERMUX_PKG_VERSION=10.2+2.0.1
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/libcdio/libcdio-paranoia-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=28d7d00e4a83d0221acda0fd2eb3e3240bf094db4c00a85998922201939fa952
+TERMUX_PKG_DEPENDS="libcdio"
+TERMUX_INSTALL_DEPS=true
```
Comment (Grimler91) [inline packages/libcdio-paranoia/build.sh:9]: Remove this

File: packages/libcdio-paranoia/build.sh [hunk 12]
```diff
@@ -0,0 +1,20 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/libcdio
+TERMUX_PKG_DESCRIPTION="This is a port of xiph.org's cdda paranoia to use libcdio for CDROM access."
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
```
Comment (Grimler91) [inline packages/libcdio-paranoia/build.sh:4]: Should be GPL-3.0

File: packages/libcdio/build.sh [hunk 13]
```diff
@@ -0,0 +1,23 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/libcdio
+TERMUX_PKG_DESCRIPTION="The GNU Compact Disc Input and Control library (libcdio) contains a library for CD-ROM and CD image access. Applications wishing to be oblivious of the OS- and device-dependent properties of a CD-ROM or of the specific details of various CD-image formats may benefit from using this library."
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="GNU Public License 3.0"
```
Comment (Grimler91) [inline packages/libcdio/build.sh:4]: Should be GPL-3.0

File: packages/cyanrip/build.sh [hunk 14]
```diff
@@ -0,0 +1,8 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/cyanreg/cyanrip
+TERMUX_PKG_DESCRIPTION="Fully featured CD ripping program able to take out most of the tedium. Fully accurate, has advanced features most rippers don't, yet has no bloat and is cross-platform."
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_LICENSE="LGPL- 2.1"
```
Comment (Grimler91) [inline packages/cyanrip/build.sh:4]: Should be LGPL-2.1

**Review by Grimler91 — CHANGES_REQUESTED** [review]

**Comment by Grimler91** [comment]
I would prefer if the commit is split into three, one for each package, to keep the commits small and git log easy to understand

File: packages/libcdio/build.sh [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+	${TERMUX_PKG_SRCDIR}/configure --prefix=${PREFIX} LDFLAGS=" -landroid-glob"
+}
+termux_step_make(){
+	cd ${TERMUX_PKG_SRCDIR} && make -j8
+}
+termux_step_make_install() {
+	cd ${TERMUX_PKG_SRCDIR} && make install
+}
+termux_step_create_debscripts() {
+	echo "echo To install python bindings use  pip3 install pycdio (Requires swig to be installed)" >> postinst
```
Comment (Grimler91) [inline packages/libcdio/build.sh:21]: The script should have a shebang as well

File: packages/libcdio/build.sh (hunk 1 above)
Comment (tathastu871) [reply packages/libcdio/build.sh]: resolved

File: packages/libcdio/build.sh (hunk 2 above)
Comment (tathastu871) [reply packages/libcdio/build.sh]: resolved

File: packages/libcdio/build.sh (hunk 4 above)
Comment (tathastu871) [reply packages/libcdio/build.sh]: resolved

File: packages/libcdio-paranoia/build.sh (hunk 10 above)
Comment (tathastu871) [reply packages/libcdio-paranoia/build.sh]: done

File: packages/libcdio-paranoia/build.sh (hunk 11 above)
Comment (tathastu871) [reply packages/libcdio-paranoia/build.sh:9]: ?? what to remove

File: packages/libcdio/build.sh (hunk 6 above)
Comment (tathastu871) [reply packages/libcdio/build.sh:8]: ??

File: packages/libcdio-paranoia/build.sh (hunk 11 above)
Comment (Grimler91) [reply packages/libcdio-paranoia/build.sh:9]: TERMUX_INSTALL_DEPS, see my other comment about it on libcdio

**Comment by tathastu871** [comment]
> I would prefer if the commit is split into three, one for each package, to keep the commits small and git log easy to understand

I dont know how to do it !new to git

**Comment by tathastu871** [comment]
Checks were failing due to unavailability of libmusicbrainz. A PR for libmusicbrainz is creted on termux-packages repo  If it is merged this checks will pass.

**Pull request closed without merging**